### DISCUSSION TITLE
[CONTRIBUTION] Speech Dataset Generator

### DISCUSSION THREAD

**Discussion by davidmartinrius** [rhasspy/piper#457] [Show and tell]
Hi everyone! 

I have just published this project on GitHub: https://github.com/davidmartinrius/speech-dataset-generator/

Now you can create datasets automatically with any audio or lists of audios.

I hope you can find it useful. 

## Here are the key functionalities of the project:

1. **Dataset Generation:** Creation of multilingual datasets with Mean Opinion Score (MOS).

2. **Silence Removal:** It includes a feature to remove silences from audio files, enhancing the overall quality.

3. **Sound Quality Improvement:** It improves the quality of the audio when needed.

4. **Audio Segmentation:** It can segment audio files within specified second ranges.

5. **Transcription:** The project transcribes the segmented audio, providing a textual representation.

6. **Gender Identification:** It identifies the gender of each speaker in the audio.

7. **Pyannote Embeddings:** Utilizes pyannote embeddings for speaker detection across multiple audio files.

8. **Automatic Speaker Naming:** Automatically assigns names to speakers detected in multiple audios.

9. **Multiple Speaker Detection:** Capable of detecting multiple speakers within each audio file.

10. **Store speaker embeddings:** The speakers are detected and stored in a Chroma database, so you do not need to assign a speaker name.

11. **Syllabic and words-per-minute metrics**

12. **Multiple input sources:** You can either use your own files or download content by pasting URLs from sources such as **YouTube**, **LibriVox** and **TED Talks**. 

Feel free to explore the project at https://github.com/davidmartinrius/speech-dataset-generator

**Comment by StoryHack** [comment]
So this sounds pretty awesome.  Thanks!

**Reply by davidmartinrius** [reply]
Thanks @StoryHack I hope you find it useful

**Comment by luiscarlos2000** [comment]
This sounds awesome as I need to organize the folders in which I'm recording the files. Could this work on Windows?
I mean, I somehow use linux, but not as a daily driver like Windows.

**Reply by davidmartinrius** [reply]
Just Linux

**Comment by luiscarlos2000** [comment]
Yikes. Unless I start my Linux VM and install that and do all the job... Even in Mac OS... really don't know about it since I'm not a regular user. But unless a Windows or MacOS version be created, I'll be for it! Hope there will be more contribution to this project. Will tell some friends to contribute as soon as they are available. This truely has a lot, a lot, a lot of potential.

**Reply by davidmartinrius** [reply]
The main reason that it is not working on Windows is this: https://github.com/davidmartinrius/speech-dataset-generator/issues/8

If you know how to solve it you will be able to work with it on Windows.

Contributions are welcome!
